src/libs/Renderer.js

```diff
@@ -41,9 +43,28 @@ export default class Renderer {
             Behaviors.edit(event);
             event.target.removeAttribute('contentEditable');
             item.querySelector('.option').classList.remove('hidden');
+            item.classList.remove('edit');
           }
         });
         this.element.appendChild(item); // updating DOM content
+        const itemList = document.querySelectorAll('#list li');
+        itemList.forEach((it) => {
+          it.addEventListener('dragstart', () => setTimeout(() => it.classList.add('dragging'), 0));
+          it.addEventListener('dragend', () => it.classList.remove('dragging'));
+        });
+
+        this.element.addEventListener('dragover', (event) => {
+          event.preventDefault();
+          const draggingItems = this.element.querySelector('li.dragging');
+          const siblings = [...this.element.querySelectorAll('li:not(.dragging)')];
+          const nextSibling = siblings.find((sibling) => {
+            const { offsetTop, offsetHeight } = sibling;
+            return event.clientY <= offsetTop + offsetHeight / 2;
+          });
+          this.element.insertBefore(draggingItems, nextSibling);
+        }, true);
+
+        this.element.addEventListener('dragenter', (event) => event.preventDefault());
```

Comment:
Awesome work here implementing the drag behavior 👏🏽 💯 this looks so nice even tho it's not updating the indexes and saving to `locaalStorage`, props to you. 🤓